c/datasetjson.c

```diff
@@ -110,6 +118,16 @@ int streamDataset(Socket *socket, char *filename, int recordLength, jsonPrinter
       }
     }
     fclose(in);
+    rcEtag = icsfDigestFinish(&digest, hash);
+
+    // Convert hash text to hex.
+    char eTag[70] = "";
```

Comment:
Is 70 enough? The hash buffer is 32 bytes, which would be 64 chars. Then 6 chars are in the tag and a null-term. That's 71. Can you ensure that even if the text changes there is enough space? Maybe something like 

```c
#define ETAG_TEXT "etag->"
char eTag[sizeof(digest.hash) * 2 + sizeof(ETAG_TEXT)] = {ETAG_TEXT};
```
The size of `eTag` should then be compile time.